sync-core/src/test/java/com/cloudant/sync/replication/ReplicatorIdTest.java

```diff
@@ -0,0 +1,119 @@
+package com.cloudant.sync.replication;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Created by tomblench on 10/12/14.
+ */
+public class ReplicatorIdTest extends ReplicationTestBase {
+
+    // source/target switched, so ids should be different
+    @Test
+    public void pullNotEqualToPush() throws URISyntaxException {
+        PullReplication pull = this.createPullReplication();
+        PushReplication push = this.createPushReplication();
+        Assert.assertNotEquals(pull.createReplicationStrategy().getReplicationId(),
+                push.createReplicationStrategy().getReplicationId());
+    }
+
+    // two pull reps identical
+    @Test
+    public void pullsEqual() throws URISyntaxException {
+        PullReplication pull1 = this.createPullReplication();
+        PullReplication pull2 = this.createPullReplication();
+        Assert.assertEquals(pull1.createReplicationStrategy().getReplicationId(),
+                pull2.createReplicationStrategy().getReplicationId());
+    }
+
+    // two push reps identical
+    @Test
+    public void pushesEqual() throws URISyntaxException {
+        PushReplication push1 = this.createPushReplication();
+        PushReplication push2 = this.createPushReplication();
+        Assert.assertEquals(push1.createReplicationStrategy().getReplicationId(),
+                push2.createReplicationStrategy().getReplicationId());
+    }
+
+    // two push reps with differing target not equal
+    @Test
+    public void pushesDifferingTargetNotEqual() throws URISyntaxException {
+        PushReplication push1 = this.createPushReplication();
+        PushReplication push2 = this.createPushReplicationDifferingTarget();
+        Assert.assertNotEquals(push1.createReplicationStrategy().getReplicationId(),
+                push2.createReplicationStrategy().getReplicationId());
+    }
+
+    // two pull reps with differing source not equal
+    @Test
+    public void pullsDifferingSourceNotEqual() throws URISyntaxException {
+        PullReplication pull1 = this.createPullReplication();
+        PullReplication pull2 = this.createPullReplicationDifferingSource();
```

Comment:
I think the differing properties should be made explicit in the test rather than hidden in methods like `DifferingSource()`. The abstraction makes it more difficult to read the tests as information relevant to the test is somewhere outside the test method.

```java
    @Test
    public void pullsDifferingSourceNotEqual() throws URISyntaxException {
        PullReplication pull1 = this.createPullReplication();
        PullReplication pull2 = this.createPullReplication();

        pull1.source = new URI("http://blah.com/");
        pull2.source = new URI("http://blah2.com/");

        Assert.assertNotEquals(pull1.createReplicationStrategy().getReplicationId(),
                pull2.createReplicationStrategy().getReplicationId());
    }
```

... and similarly for the other tests.